packages/nx/src/executors/utils/convert-nx-executor.ts

```diff
@@ -51,39 +51,43 @@ function toObservable<T extends { success: boolean }>(
 ): Observable<T> {
   return new (require('rxjs') as typeof import('rxjs')).Observable(
     (subscriber) => {
-      promiseOrAsyncIterator.then((value) => {
-        if (!(value as any).next) {
-          subscriber.next(value as T);
-          subscriber.complete();
-        } else {
-          let asyncIterator = value as AsyncIterableIterator<T>;
+      promiseOrAsyncIterator
+        .then((value) => {
+          if (!(value as any).next) {
+            subscriber.next(value as T);
+            subscriber.complete();
+          } else {
+            let asyncIterator = value as AsyncIterableIterator<T>;
 
-          function recurse(iterator: AsyncIterableIterator<T>) {
-            iterator
-              .next()
-              .then((result) => {
-                if (!result.done) {
-                  subscriber.next(result.value);
-                  recurse(iterator);
-                } else {
-                  if (result.value) {
+            function recurse(iterator: AsyncIterableIterator<T>) {
+              iterator
+                .next()
+                .then((result) => {
+                  if (!result.done) {
                     subscriber.next(result.value);
+                    recurse(iterator);
+                  } else {
+                    if (result.value) {
+                      subscriber.next(result.value);
+                    }
+                    subscriber.complete();
                   }
-                  subscriber.complete();
-                }
-              })
-              .catch((e) => {
-                subscriber.error(e);
-              });
-          }
+                })
+                .catch((e) => {
+                  subscriber.error(e);
+                });
+            }
 
-          recurse(asyncIterator);
+            recurse(asyncIterator);
 
-          return () => {
-            asyncIterator.return();
-          };
-        }
-      });
+            return () => {
+              asyncIterator.return();
+            };
+          }
+        })
+        .catch((err) => {
+          subscriber.error(err);
+        });
```

Comment:
This PR got hard to read because of formatting, this block is the important bit.